src/Observable.ts

```diff
@@ -13,9 +13,13 @@ export interface Subscribable<T> {
   subscribe(observer: Observer<T>): Subscription;
 }
 
-export type SubscribableOrPromise<T> = Subscribable<T> | Promise<T>;
-export type ArrayOrIterator<T> = Iterator<T> | ArrayLike<T>;
-export type ObservableInput<T> = SubscribableOrPromise<T> | ArrayOrIterator<T>;
+export interface Observablesque<T> {
+  [Symbol.observable](): Subscribable<T>;
```

Comment:
Small thing here. `[Symbol.observable]()` should return a type that is `Subscribable<T> | Observablesque<T>` at a minimum. (Similar to how Iterators have a `Symbol.iterator` on them)